tools/build_project.py

```diff
@@ -169,6 +169,20 @@ def load_toolchains(paths: list[pathlib.Path]) -> dict[pathlib.Path, str]:
     return toolchains
 
 
+def subprocess_run_verbose(command: list[str], prefix: str) -> None:
+    result = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+
+    with result.stdout:
+        for line in iter(result.stdout.readline, b""):
```

Comment:
You should be able to use context managers and iterators to avoid the unnecessary `wait` and to avoid calling `readline` explicitly:

```python
with subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.STDOUT) as proc:
    for line in proc.stdout:
        LOGGER.info("[%s] %r", prefix, line.decode("utf-8").strip())

if proc.returncode != 0:
    LOGGER.error("[%s] Error: %s", prefix, proc.returncode)
    sys.exit(1)
```